dashboard/src/components/OperatorInstance/OperatorInstance.tsx

```diff
@@ -93,6 +125,28 @@ class OperatorInstance extends React.Component<IOperatorInstanceProps> {
       </CardGrid>
     );
   };
+
+  private openModal = () => {
+    this.setState({
+      modalIsOpen: true,
+    });
+  };
+
+  private closeModal = async () => {
+    this.setState({
+      modalIsOpen: false,
+    });
+  };
+
+  private handleDeleteClick = async () => {
+    const { namespace, resource } = this.props;
+    const { crd } = this.state;
+    const deleted = await this.props.deleteResource(namespace, crd!.name.split(".")[0], resource!);
+    this.closeModal();
+    if (deleted) {
+      this.props.push(`/apps/ns/${namespace}`);
```

Comment:
Seems odd to be pushing down `push` as a function to the props (it'd be like sending `dispatch` as a prop to the component, rather than using `dispatch` at the controller. I'd expect instead to see a function `deleted()` which is provided as a prop (or `onDelete` or whatever) which handles the `push` at the controller. But not a big deal.

Reply:
we are using the same approach in different places, that's why I used it:

```
src/containers/HeaderContainer/HeaderContainer.ts
37:    push: (path: string) => dispatch(push(path)),

src/containers/AppNewContainer/AppNewContainer.tsx
47:    push: (location: string) => dispatch(push(location)),

src/containers/AppViewContainer/AppViewContainer.tsx
39:    push: (location: string) => dispatch(push(location)),

src/containers/AppUpgradeContainer/AppUpgradeContainer.tsx
56:    push: (location: string) => dispatch(push(location)),

src/containers/ServiceClassViewContainer/ServiceClassViewContainer.ts
50:    push: (location: string) => dispatch(push(location)),
```

In any case, what do you mean by `controller`? Why are you comparing `push` to `dispatch`?

Reply:
Generally I thought we try to keep components being dumb - they just react to UI events by either setting props or calling a function like, in this case, `deleted()`, without knowing what it does, so that something else which controls the component (ie. the thing with the brains :) ) can take the appropriate action. But no big deal, just unexpected.